File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 5/29/2026, 12:01:27 PM
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if s=="":
            return 0
        map=set()
        max_len=float('-inf')
        left=0
        for right in range(len(s)):
            while s[right] in map:
                map.remove(s[left])
                left+=1
            map.add(s[right])
            max_len=max(max_len,right-left+1)
        return max_len

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 5/29/2026, 12:01:33 PM
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int left=0;
        HashSet<Character> hs=new HashSet<>();
        int max_sum=0;
        for(int right=0;right<s.length();right++){
            while(hs.contains(s.charAt(right))){
                hs.remove(s.charAt(left));
                left++;
            }
            hs.add(s.charAt(right));
            max_sum=Math.max(max_sum,right-left+1);
        }
        return max_sum;

    }
}